src/main/scala/com/fulcrumgenomics/bam/FilterBam.scala

```diff
@@ -51,6 +52,11 @@ import htsjdk.samtools.util.IntervalList
      |marked as secondary alignments, reads marked as duplicates, and if a set of Intervals are provided,
      |reads that do not overlap any of the intervals.
      |
+     |The `--remove-unpaired-reads`, which removes reads that are not part of a mapped pair will remove
```

Comment:
Can you change `--remove-unpaired-reads` to `--remove-unmapped-pairs`?  I thought it was named to remove fragment reads (does it)?  Perhaps you want to have an option `--keep-only-mapped-pairs` instead.